docs/infrastructure/microk8s-quickstart.md

```diff
@@ -0,0 +1,136 @@
+# Setting up a microk8s instance
+
+[Microk8s](https://microk8s.io/) is a single-package fully conformant lightweight Kubernetes that works on 42 Linux versions. The project is publicly available and can be [found here](https://github.com/ubuntu/microk8s).
+
+Follow the steps below to set up a microk8s instance from scratch, including all the necessary dependencies and configuration.
+
+## 1. Concepts
+
+### Kubeconfig
+
+Any tool that operates on top of a cluster needs to know which cluster it is going to work on. This execution context is typically called the `kubeconfig` which is stored in the `~/.kube/config` file. This file basically defines a list of clusters and it holds several pieces of information per cluster entry:
+
+* The address of the cluster (e.g. https://104.248.34.242:164309).
+* The user friendly name of the cluster (e.g. microk8s-instance).
+* A reference to the current context. This is the active cluster against which any kubernetes control (`kubectl`) command you run will get executed.
+
+### Helm and tiller
+
+Two executables will get installed onto the cluster as part of this process: `helm` and `tiller`.
+
+* `helm` is responsible for resolving the configuration of the chart to be installed, while issuing the correct install commands onto the cluster.
+* `tiller` is responsible for receiving the install commands issued by `helm`, as well as managing the lifecycle of the components that have been installed.
+
+`helm` is the client facing side, while `tiller` is the server/cluster facing side.
+
+## 2. Prepare your environment
+
+Prepare your environment to set up the microk8s cluster.
+For your infrastructure, you will need the following:
+
+1. A machine running Ubuntu 18.04 LTS.
+2. A postgres database and open firewall rule. ([Documentation can be found here.](../requirements.md))
+
+To start installing Codacy, you will need to:
+
+1. Establish a session (local or remote) onto the machine described in 1. above.
+
+Assuming that you are starting from a blank slate, the first step is to install microk8s. Otherwise, consider that Codacy is supported up to kubernetes 1.15. Please jump on to the section [Configuring microk8s](###4.-Configuring-microk8s) if you are on a compatible kubernetes version.
+
+## 3. Installing microk8s
```

Comment:
suggestion from rodrigo:
Single script with verification at the begging if we are on the right OS, kernel, etc